src/appleseed/renderer/modeling/object/rectangleobject.cpp

```diff
@@ -143,6 +143,21 @@ double RectangleObject::get_uncached_height() const
     return m_params.get_optional<double>("height", 1.0);
 }
 
+void RectangleObject::get_origin_and_axes(

```

Comment:
The implementation of this method doesn't appear to match the description from the header file:

    The rectangle is centered at the origin and lies in the X-Z plane.

If that was the case, `origin` would always be (0, 0, 0), no?